src/core/grid_based_algorithms/lb_interpolation.cpp

```diff
@@ -1,17 +1,39 @@
+#include <boost/mpi/collectives.hpp>
+
 #include "communication.hpp"
 #include "config.hpp"
 #include "grid.hpp"
 #include "grid_based_algorithms/lattice.hpp"
+#include "grid_based_algorithms/lb_interpolation.hpp"
 #include "utils/Vector.hpp"
 
 #include "lb.hpp"
 #include "lb_interface.hpp"
 #include "lbgpu.hpp"
 
+namespace {
+
+InterpolationOrder interpolation_order = InterpolationOrder::linear;
+}
+
+void mpi_set_interpolation_order_slave(int, int) {
+  boost::mpi::broadcast(comm_cart, interpolation_order, 0);
+}
+
 #if defined(LB) || defined(LB_GPU)
 
-namespace {
+void lb_lbinterpolation_set_interpolation_order(
+    InterpolationOrder const &order) {
+  interpolation_order = order;
+  boost::mpi::broadcast(comm_cart, interpolation_order, 0);
```

Comment:
The mpi calls need to be the other way around, this is a deadlock.

Reply:
Could you please explain?

Reply:
You are blocking on ` boost::mpi::broadcast` before the callback are invoked.